### DISCUSSION TITLE
Can MatLabReader use a URL as the file location of the .mat file?

### DISCUSSION THREAD

**Discussion by ghost** [mathnet/mathnet-numerics#788] [Ideas]
I have the following code, which works fine:
String filePath = "C:\MyFolder\MyMatFile.mat";
Matrix<double> m = MatlabReader.Read<double>(filePath);

However when I change filePath to a URL, I get the following error:
String filePath = "https://myproj.blob.core.windows.net/mycontainer/MyMatFile.mat";
Matrix<double> m = MatlabReader.Read<double>(filePath);

Error: System.NotSupportedException: The given path's format is not supported.

**Comment by ghost** [comment]
Here's my code:

using System;
using MathNet.Numerics.LinearAlgebra;
using MathNet.Numerics.Data.Matlab;

namespace TestMatlabAzure
{
    class Program
    {
        // this one works
        private const string filePath = "C:\\Data\\ImportFiles\\CMI\\Test_20190615\\probFile_nn07c_20180316_000000_0.mat";
        
        // this one doesn't
        //private const string filePath = "https://elpround3.blob.core.windows.net/test0705/probFile_nn07c_20180316_000000_0.mat";

        static void Main(string[] args)
        {
            Console.WriteLine("Reading .mat file....");
            Matrix<double> m = MatlabReader.Read<double>(filePath);
            Console.WriteLine(m.ToString());
            Console.ReadKey();
        }
    }
}

**Comment by cdrnet** [comment]
While I do understand that this could be convenient, to be honest I don't think a math library should be in the business of opening any network connections. From this point of view, actually even the method accepting a file path (instead of just a stream or reader) was maybe design mistake. Although maybe justifiable due to low risk/effort/complexity - neither of which would be true for networking though.
